app/services/push_notification/http_push.rb

```diff
@@ -0,0 +1,72 @@
+module PushNotification
+  # Delivers SETs (Security Event Tokens) via the HTTP Push protocol
+  # https://tools.ietf.org/html/draft-ietf-secevent-http-push-00
+  class HttpPush
+    include Rails.application.routes.url_helpers
+
+    attr_reader :event
+
+    # shorthand for creating an instance then calling #deliver, easier to stub
+    def self.deliver(event, now: Time.zone.now)
+      new(event, now: now).deliver
+    end
+
+    def initialize(event, now: Time.zone.now)
+      @event = event
+      @now = now
+    end
+
+    def deliver
+      event.user.
+        service_providers.
+        merge(Identity.not_deleted).
+        with_push_notification_urls.each do |service_provider|
+          deliver_one(service_provider)
+        end
+    end
+
+    private
+
+    attr_reader :now
+
+    def deliver_one(service_provider)
+      response = faraday.post(
+        service_provider.push_notification_url,
+        jwt(service_provider),
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/secevent+jwt',
+      )
+
+      unless response.success?
```

Comment:
Other than adding a way to retry failed pushes (or at least store the failed ones like push for account delete) the PR LGTM.

Reply:
Ok do you think it's important to add that in this PR? Or is that an OK follow-up candidate?

Reply:
I think the rationale was if we hadn't hooked up the retries at least we could see the packets that failed and eventually deliver them.  I guess it all depends on when we think this will go live.  I'll go ahead and approve...